### PR TITLE
fix: fix player argument parser not getting all players

### PR DESCRIPTION
**Pull request by Wyvest**
## Description
Currently, we get all the players in `Minecraft.theWorld` and filter out of that. However, this does not return all players in the server (for example, if we hid players in the Hypixel lobby using the default Hypixel thing it won't show non-hidden players in the tab-completion) and we return `EntityPlayer`, which is mostly unnecessary for commands (which most of the time only want the player's name or UUID).

With this PR, we return `GameProfile` instead and use a different method of filtering players.

## Checklist
- [x] I made a clear description of what was changed
- [x] I stated why these changes were necessary
- [x] I updated documentation or said what needs to be updated
- [x] I made sure these changes are backwards compatible
- [x] This pull request is for one feature/bug fix

### CODE REVIEW

**Review by xtrm-en — CHANGES_REQUESTED** [review]
For retro compatibility's sake, let's keep the old parser named `PlayerArgumentParser` and rename the new `PlayerProfileArgumentParser` or `NewPlayerArgumentParser`?

**Comment by Wyvest** [comment]
> For retro compatibility's sake, let's keep the old parser named `PlayerArgumentParser` and rename the new `PlayerProfileArgumentParser` or `NewPlayerArgumentParser`?

no mods touch it directly so...

**Comment by xtrm-en** [comment]
> no mods touch it directly so...
then why mark it as deprecated instead of removing it altogether?

File: versions/src/main/java/cc/polyfrost/oneconfig/utils/commands/arguments/OldPlayerArgumentParser.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+import java.lang.reflect.Parameter;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * The old player argument parser. Returns an {@link EntityPlayer}.
+ * @deprecated Use {@link PlayerArgumentParser} instead.
+ */
+@Deprecated
+public class OldPlayerArgumentParser extends ArgumentParser<EntityPlayer> {
```
Comment (nxtdaydelivery) [inline versions/src/main/java/cc/polyfrost/oneconfig/utils/commands/arguments/OldPlayerArgumentParser.java:20]: call this entity player argument parser please

File: versions/src/main/java/cc/polyfrost/oneconfig/utils/commands/arguments/PlayerArgumentParser.java [hunk 2]
```diff
@@ -1,33 +1,71 @@
...
+    public GameProfile parse(@NotNull String arg) {
+        List<GameProfile> matchingPlayers = getMatchingPlayers(arg, false);
+        for (GameProfile profile : matchingPlayers) {
+            return profile;
+        }
+        return new GameProfile(getUUID(arg), arg);
     }
 
-    public static List<EntityPlayer> getMatchingPlayers(String arg) {
+    // checks mojang api for player uuid from name
```
Comment (nxtdaydelivery) [inline versions/src/main/java/cc/polyfrost/oneconfig/utils/commands/arguments/PlayerArgumentParser.java:33]: I'll be real why do we have this it is just a waste of processing time and memory they can implement this themselves if they want

**Review by nxtdaydelivery — COMMENTED** [review]
hm.

File: versions/src/main/java/cc/polyfrost/oneconfig/utils/commands/arguments/PlayerArgumentParser.java (hunk 2 above)
Comment (Wyvest) [reply versions/src/main/java/cc/polyfrost/oneconfig/utils/commands/arguments/PlayerArgumentParser.java:33]: so they can put nonexistant players if they want (for stuff like hytils gexp checker)

File: versions/src/main/java/cc/polyfrost/oneconfig/utils/commands/arguments/OldPlayerArgumentParser.java (hunk 1 above)
Comment (Wyvest) [reply versions/src/main/java/cc/polyfrost/oneconfig/utils/commands/arguments/OldPlayerArgumentParser.java:20]: Fine

**Review by xtrm-en — APPROVED** [review]
lgtm

**Review by pauliesnug — APPROVED** [review]
can you quality control me 🥺

**Pull request merged**